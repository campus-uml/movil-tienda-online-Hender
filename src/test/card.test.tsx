import { describe, it, expect } from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import RecipeCard from '../components/ui/card';

describe('Testing de RecipeCard.tsx', () => {
    it('Renderiza correctamente el componente', () => {
        const { baseElement } = render(<RecipeCard />);
        expect(baseElement).toMatchSnapshot();
    });

    it('Debe renderizar el título "Online Shopping"', () => {
        const { getByText } = render(<RecipeCard />);
        expect(getByText(/Online Shopping/i)).toBeTruthy();
    });

    it('Debe renderizar la descripción correctamente', () => {
        const { getByText } = render(<RecipeCard />);
        expect(getByText(/Shop anytime, anywhere, and enjoy the best deals from the comfort of your home./i)).toBeTruthy();
    });

    it('Debe renderizar el botón "Shop Now"', () => {
        const { getByText } = render(<RecipeCard />);
        expect(getByText(/Shop Now/i)).toBeTruthy();
    });

    it('Debe reaccionar correctamente al hacer clic en el botón', () => {
        const { getByText } = render(<RecipeCard />);
        const button = getByText(/Shop Now/i);
        fireEvent.click(button);
        expect(button).toBeTruthy(); // Aquí puedes agregar más lógica si el botón debe realizar alguna acción
    });
});
