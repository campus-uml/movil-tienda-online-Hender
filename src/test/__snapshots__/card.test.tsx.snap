// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing de RecipeCard.tsx > Renderiza correctamente el componente 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-center min-h-screen bg-gradient-to-b from-blue-50 via-white to-blue-100 px-4"
    >
      <div
        class="bg-white p-8 lg:p-12 rounded-3xl shadow-xl max-w-md w-full text-center transform hover:scale-105 transition-transform duration-300"
      >
        <img
          alt="Online Shopping"
          class="w-44 h-44 lg:w-52 lg:h-52 mx-auto rounded-xl object-cover shadow-md"
          src="/src/components/ui/images/imagen1.png"
        />
        <h2
          class="text-3xl lg:text-4xl font-extrabold text-gray-900 mt-6"
        >
          Online Shopping
        </h2>
        <p
          class="text-gray-500 text-base lg:text-lg mt-3"
        >
          Shop anytime, anywhere, and enjoy the best deals from the comfort of your home.
        </p>
        <button
          class="mt-6 w-[90%] bg-green-500 hover:bg-green-600 text-white font-semibold py-3 px-6 rounded-xl shadow-md hover:shadow-lg transition-all duration-300"
        >
          Shop Now
        </button>
      </div>
    </div>
  </div>
</body>
`;
